httpio.py

```diff
@@ -92,12 +90,12 @@ def read(self, size=-1):
                         self.block_size * (idx + 1)]
 
             data = []
-            for idx in range(sector0, sector1):
+            for idx in range(sector0, sector1 + 1):
                 start = offset0 if idx == sector0 else None
-                end   = offset1 if idx == sector1 else None
+                end = offset1 if idx == sector1 else None
```

Comment:
Thanks for spotting and fixing this.

There's not much in this, but I think it's a bit nicer to change this line to read:

```
end = offset1 if idx == (sector1 - 1) else None
```

... rather than change the value of `sector1` up-front. I suggest this because currently sector0 and sector1 represent a half-open interval, which is what `range()` accepts and is generally [easier to reason about](http://www.cs.utexas.edu/users/EWD/ewd08xx/EWD831.PDF).